### ISSUE TITLE
Add keyboard controls

### ISSUE THREAD

**Issue by JodiTheTigger** [passcod/mechanicalbird#5]
When you type in a new description it acts as if you pressed stop/start and started typing. Allow the user to cancel if they change their mind.

Just a thought.

**Comment by passcod** [comment]
What browser? That's a bug

**Comment by JodiTheTigger** [comment]
linux + firefox + noscript + https everywhere + privacy badger

So to calrify: are you saying that if I just start typing, the page should automatically stop the current task, start a new one, and then enter the text as the description of the new one?

**Comment by passcod** [comment]
Ah. No. Start/stop is only done by the button press. Editing descriptions is completely independent of that. The idea is you hit the button, and then you can add a description later. Soon later, or much later, in fact. You can totally enter descriptions for all tasks at the end of your day, and leave the tasks named with the default until then, if you wish to.

There's no way (currently?) to "change one's mind." If you press stop by accident, press start again and name it the same. If you press start by accident, press stop again and name the resulting "empty" task as such, like "disregard, mistake" or something.

The entire app is literally three actions: start, stop, and edit a description.

If you "just start typing," behaviour will vary depending on what is focused at the moment. If a task description is focused, you'll edit that. If the start/stop button is focused, it will behave as per normal button/keyboard semantics, which probably means that if/when you hit enter it will treat it as a button press. There's no special handling.

**Comment by JodiTheTigger** [comment]
Ah, ok. Ta.

Then I guess this is a feature request/idea then :-)

**Comment by passcod** [comment]
Would it work if I made it so, say, Ctrl-Enter triggered "start a new task", and immediately focused the new task so you could type the name directly?

**Comment by JodiTheTigger** [comment]
Ah, the dangers of feature creep.

Yes that could work, but I would like it to be even simpler for the user
(_ahem_ me _ahem_).

So I would imagine it like this. Press play. then when you did something
you would just click on the website and start typing what you just did. The
website would auto stop/start the timer, and direct what you're typing into
the description of the recently completed time block.

No pressing ctrl-enter, just typing. :-)

again, just a thought, I don't use time tracking, so please don't spend
time on this if you don't feel like it would be useful for yourself.

**Comment by passcod** [comment]
Yeah, that can't work, cause we can't reliably detect whether you're typing in a task to edit it, or typing somewhere random to create one, or you just forgot which tab you opened, or you clicked on a task but not on the text and you're expecting it to edit, or... Ctrl-Enter + autofocus, takes slightly longer, sure, but look at it as a compromise: it has the advantage of being a single mapping of a single key combination to a single action and then letting the browser handle input.

Also, that way Tab-navigation still works.
